test/gyppies-test.js

```diff
@@ -26,7 +32,9 @@ function build(name) {
     const argvFile = path.join(folder, 'argv.json');
     const argv = fs.existsSync(argvFile) ? require(argvFile) : [];
 
-    let p = spawnSync(process.execPath, [ gyp ].concat(argv), spawnOpts);
+    let p = spawnSync(process.execPath,
+                      (process.env.running_under_istanbul ? istanbulArgs : [])
+                      .concat([ gyp ]).concat(argv), spawnOpts);
```

Comment:
Let's do it a bit differently. Above: `const argv` => `let argv`;

Then before this line:

```
argv.push(gyp);
if (process.env.running_under_istanbul)
  argv = istanbulArgs.concat(argv);
```

And pass just `argv` to the `spawnSync`.